scheduler can fail until restart if temporal isn't ready on startup

## Expected Behavior
The scheduler should attempt to start, wait for temporal to start, and then begin running.

## Current Behavior
In some (presumably rare) cases, the scheduler can fail within `waitForTemporalServerAndLog` with an exception like:
```
[32mairbyte-scheduler   |[0m Exception in thread "main" io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
[32mairbyte-scheduler   |[0m 	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
[32mairbyte-scheduler   |[0m 	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
[32mairbyte-scheduler   |[0m 	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
[32mairbyte-scheduler   |[0m 	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.listNamespaces(WorkflowServiceGrpc.java:2588)
[32mairbyte-scheduler   |[0m 	at io.airbyte.workers.temporal.TemporalPool.getNamespaces(TemporalPool.java:105)
[32mairbyte-scheduler   |[0m 	at io.airbyte.workers.temporal.TemporalPool.waitForTemporalServerAndLog(TemporalPool.java:84)
[32mairbyte-scheduler   |[0m 	at io.airbyte.workers.temporal.TemporalPool.run(TemporalPool.java:58)
[32mairbyte-scheduler   |[0m 	at io.airbyte.scheduler.app.SchedulerApp.start(SchedulerApp.java:113)
[32mairbyte-scheduler   |[0m 	at io.airbyte.scheduler.app.SchedulerApp.main(SchedulerApp.java:230)
[32mairbyte-scheduler   |[0m Caused by: io.grpc.netty.shaded.io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: airbyte-temporal/172.23.0.6:7233
[32mairbyte-scheduler   |[0m Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.channel.unix.Errors.throwConnectException(Errors.java:124)
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.channel.unix.Socket.finishConnect(Socket.java:243)
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:672)
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:649)
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:529)
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:465)
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[32mairbyte-scheduler   |[0m 	at io.grpc.netty.shaded.io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
```

## Severity of the bug for you
Medium

This happened multiple times for this person on restarting from scratch. It probably runs with unfortunate timing on his machine. I'm pretty sure the general solution is to just swallow exceptions during the namespace check (`waitForTemporalServerAndLog`) so the retries can continue until it succeeds. The instance we saw of this issue was "resolved" by running `docker restart airbyte-scheduler`.

Relevant Slack thread: https://airbytehq.slack.com/archives/C01MFR03D5W/p1618420149490500

--- a/airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalPool.java
+++ b/airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalPool.java
@@ -84,12 +84,21 @@ public class TemporalPool implements Runnable {
     factory.start();
   }
 
-  private void waitForTemporalServerAndLog() {
+  protected void waitForTemporalServerAndLog() {
     LOGGER.info("Waiting for temporal server...");
 
-    while (!getNamespaces(temporalService).contains("default")) {
+    boolean temporalStatus = false;
+
+    while (!temporalStatus) {
       LOGGER.warn("Waiting for default namespace to be initialized in temporal...");
       wait(2);
+
+      try {
+        temporalStatus = getNamespaces(temporalService).contains("default");
+      } catch (Exception e) {
+        // Ignore the exception because this likely means that the Temporal service is still initializing.
+        LOGGER.warn("Ignoring exception while trying to request Temporal namespaces:", e);
+      }
     }
 
     // sometimes it takes a few additional seconds for workflow queue listening to be available
@@ -106,7 +115,7 @@ public class TemporalPool implements Runnable {
     }
   }
 
-  private static Set<String> getNamespaces(WorkflowServiceStubs temporalService) {
+  protected static Set<String> getNamespaces(WorkflowServiceStubs temporalService) {
     return temporalService.blockingStub()
         .listNamespaces(ListNamespacesRequest.newBuilder().build())
         .getNamespacesList()

--- a/airbyte-workers/src/test/java/io/airbyte/workers/temporal/TemporalPoolTest.java
+++ b/airbyte-workers/src/test/java/io/airbyte/workers/temporal/TemporalPoolTest.java
@@ -1,36 +1,60 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
 package io.airbyte.workers.temporal;
-
-import io.airbyte.workers.process.ProcessBuilderFactory;
-import io.temporal.api.namespace.v1.NamespaceInfo;
-import io.temporal.api.workflowservice.v1.DescribeNamespaceResponse;
-import io.temporal.serviceclient.WorkflowServiceStubs;
-import org.apache.commons.compress.utils.Lists;
-import org.junit.jupiter.api.Test;
-import org.mockito.Mockito;
-
-import java.nio.file.Path;
-import java.util.List;
 
 import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-class TemporalPoolTest {
-    @Test
-    public void testWaitForTemporalServerOnLogThrowsException() {
-        WorkflowServiceStubs workflowServiceStubs = mock(WorkflowServiceStubs.class, Mockito.RETURNS_DEEP_STUBS);
-        Path path = mock(Path.class);
-        ProcessBuilderFactory processBuilderFactory = mock(ProcessBuilderFactory.class);
-        TemporalPool testPool = new TemporalPool(workflowServiceStubs, path, processBuilderFactory);
-        DescribeNamespaceResponse describeNamespaceResponse = mock(DescribeNamespaceResponse.class);
-        NamespaceInfo namespaceInfo = mock(NamespaceInfo.class);
+import io.airbyte.workers.process.ProcessBuilderFactory;
+import io.temporal.api.namespace.v1.NamespaceInfo;
+import io.temporal.api.workflowservice.v1.DescribeNamespaceResponse;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
 
-        when(namespaceInfo.getName()).thenReturn("default");
-        when(describeNamespaceResponse.getNamespaceInfo()).thenReturn(namespaceInfo);
-        when(workflowServiceStubs.blockingStub().listNamespaces(any()).getNamespacesList())
-                .thenThrow(RuntimeException.class)
-                .thenReturn(List.of(describeNamespaceResponse));
-        testPool.waitForTemporalServerAndLog();
-    }
+class TemporalPoolTest {
+
+  @Test
+  public void testWaitForTemporalServerOnLogThrowsException() {
+    WorkflowServiceStubs workflowServiceStubs = mock(WorkflowServiceStubs.class, Mockito.RETURNS_DEEP_STUBS);
+    Path path = mock(Path.class);
+    ProcessBuilderFactory processBuilderFactory = mock(ProcessBuilderFactory.class);
+    TemporalPool testPool = new TemporalPool(workflowServiceStubs, path, processBuilderFactory);
+    DescribeNamespaceResponse describeNamespaceResponse = mock(DescribeNamespaceResponse.class);
+    NamespaceInfo namespaceInfo = mock(NamespaceInfo.class);
+
+    when(namespaceInfo.getName()).thenReturn("default");
+    when(describeNamespaceResponse.getNamespaceInfo()).thenReturn(namespaceInfo);
+    when(workflowServiceStubs.blockingStub().listNamespaces(any()).getNamespacesList())
+        .thenThrow(RuntimeException.class)
+        .thenReturn(List.of(describeNamespaceResponse));
+    testPool.waitForTemporalServerAndLog();
+  }
+
 }

--- a/airbyte-workers/src/test/java/io/airbyte/workers/temporal/TemporalPoolTest.java
+++ b/airbyte-workers/src/test/java/io/airbyte/workers/temporal/TemporalPoolTest.java
@@ -1,0 +1,36 @@
+package io.airbyte.workers.temporal;
+
+import io.airbyte.workers.process.ProcessBuilderFactory;
+import io.temporal.api.namespace.v1.NamespaceInfo;
+import io.temporal.api.workflowservice.v1.DescribeNamespaceResponse;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import org.apache.commons.compress.utils.Lists;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class TemporalPoolTest {
+    @Test
+    public void testWaitForTemporalServerOnLogThrowsException() {
+        WorkflowServiceStubs workflowServiceStubs = mock(WorkflowServiceStubs.class, Mockito.RETURNS_DEEP_STUBS);
+        Path path = mock(Path.class);
+        ProcessBuilderFactory processBuilderFactory = mock(ProcessBuilderFactory.class);
+        TemporalPool testPool = new TemporalPool(workflowServiceStubs, path, processBuilderFactory);
+        DescribeNamespaceResponse describeNamespaceResponse = mock(DescribeNamespaceResponse.class);
+        NamespaceInfo namespaceInfo = mock(NamespaceInfo.class);
+
+        when(namespaceInfo.getName()).thenReturn("default");
+        when(describeNamespaceResponse.getNamespaceInfo()).thenReturn(namespaceInfo);
+        when(workflowServiceStubs.blockingStub().listNamespaces(any()).getNamespacesList())
+                .thenThrow(RuntimeException.class)
+                .thenReturn(List.of(describeNamespaceResponse));
+        testPool.waitForTemporalServerAndLog();
+    }
+}

